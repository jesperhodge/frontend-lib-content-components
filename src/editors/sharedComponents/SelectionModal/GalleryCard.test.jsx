import React from 'react';
import { shallow } from 'enzyme';

import { Image } from '@edx/paragon';
import { GalleryCard } from './GalleryCard';

describe('GalleryCard component', () => {
  const img = {
    externalUrl: 'props.img.externalUrl',
    displayName: 'props.img.displayName',
    dateAdded: 12345,
  };
  let el;
  beforeEach(() => {
    el = shallow(<GalleryCard asset={img} />);
  });
  test(`snapshot: dateAdded=${img.dateAdded}`, () => {
    expect(el).toMatchSnapshot();
  });
  it('loads Image with src from image external url', () => {
    expect(el.find(Image).props().src).toEqual(img.externalUrl);
  });
});
