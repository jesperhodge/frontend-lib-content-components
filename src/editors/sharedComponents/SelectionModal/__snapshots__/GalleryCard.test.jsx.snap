// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryCard component snapshot: dateAdded=12345 1`] = `
<SelectableBox
  className="card bg-white"
  key="props.img.externalUrl"
  style={
    Object {
      "padding": "10px 20px",
    }
  }
  type="radio"
>
  <div
    className="card-div d-flex flex-row flex-nowrap"
  >
    <div
      style={
        Object {
          "height": "100px",
          "margin": "16px 0 0 0",
          "position": "relative",
          "width": "200px",
        }
      }
    >
      <Image
        src="props.img.externalUrl"
        style={
          Object {
            "height": "100px",
            "width": "200px",
          }
        }
      />
    </div>
    <div
      className="card-text p-3"
    >
      <h3>
        props.img.displayName
      </h3>
      <p
        style={
          Object {
            "fontSize": "11px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Added {date} at {time}"
          description="File date-added string"
          id="authoring.selectionmodal.addedDate.label"
          values={
            Object {
              "date": <FormattedDate
                value={12345}
              />,
              "time": <FormattedTime
                value={12345}
              />,
            }
          }
        />
      </p>
    </div>
  </div>
</SelectableBox>
`;
